app/services/MessageParser.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import model.Message
+
+trait MessageParser {
+
+  def parse(message: Message): ParsingResult
+}
+
+sealed trait ParsingResult
+case class FileUploadedEvent(bucket: String, objectKey: String) extends ParsingResult
+case class UnsupportedMessage(reason: String) extends ParsingResult
```

Comment:
As discussed, could be changed to:
sealed trait ParsingResult
case class SuccessfulEventParsingResult(bucket: String, objectKey: String)
case class FailedEventParsingResult(reason: String)

We can then look later at introducing Try or Either.